### PR TITLE
Update Module documentation

### PR DESCRIPTION
**Pull request by bzsolt**
IoT.js-DCO-1.0-Signed-off-by: Zsolt Borbély zsborbely.u-szeged@partner.samsung.com

### CODE REVIEW

**Comment by bzsolt** [comment]
This isn't a WIP commit, I've updated the PR, please check it.

File: docs/api/IoT.js-API-Module.md [hunk 1]
```diff
@@ -2,48 +2,40 @@
...
 
-## `require('id')` works as follows:
-For each directory in search paths above,
+`require` function searches for modules in the following order:
 
-0. If native module named 'id' exists, load it and return.
-1. If id.js file exists, load it and return.
-2. If a directory id exists, load it as an package and return.
+1. Current working directory.
+2. `node_modules` directory under current working directory.
```
Comment (daeyeon) [inline docs/api/IoT.js-API-Module.md]: please rename this to `iotjs_modules`, and modify the corresponding unit tests.

File: docs/api/IoT.js-API-Module.md [hunk 2]
```diff
@@ -2,48 +2,40 @@
...
+3. `$HOME/node_modules`
+4. `$NODE_PATH/node_modules`
```
Comment (daeyeon) [inline docs/api/IoT.js-API-Module.md]: Pleas rename `NODE_PATH` to `IOTJS_PATH` , modify the related sources, and verify them.

File: docs/api/IoT.js-API-Module.md [hunk 3]
```diff
@@ -2,48 +2,40 @@
...
 
-## `require('id')` as a package
-For a given id, if a directory id exists, module system consider the directory as a package.
+For each directory in search paths above:
 
-0. if 'id/node_modules/package.json' contains 'main' property, load the file named main property.
-1. if either id/node_modules/package.json or main property exists, load index.js
+- If a native module named **id** exists, load it and return.
```
Comment (bzsolt) [inline docs/api/IoT.js-API-Module.md]: I mean which module came from the IoT.js itself. (but this sentence currently exists in the document, I've just updated the style)

File: docs/api/IoT.js-API-Module.md (hunk 1 above)
Comment (bzsolt) [inline docs/api/IoT.js-API-Module.md]: I've opened a PR for this (https://github.com/Samsung/iotjs/pull/919)

File: docs/api/IoT.js-API-Module.md (hunk 3 above)
Comment (zherczeg) [inline docs/api/IoT.js-API-Module.md]: Is this a higher priority than 1. Current working directory. ?

File: docs/api/IoT.js-API-Module.md [hunk 4]
```diff
@@ -2,48 +2,40 @@
...
-
-- [Module](#module)
-    - [Module Functions](#module-functions)
-        - [`require(id)`](#requireid)
-    - [Module Loading by `require` function](#module-loading-by-require-function)
-
 # Module
-`module` implements `require` function.
+This module implements `require` function.
 Since `require` function is implicitly passed to user code, user do not need to import it explicitly.
```
Comment (zherczeg) [inline docs/api/IoT.js-API-Module.md]: I would rephrase this to:

The `require` function is always available there is no need to import `module` explicitly.

File: docs/api/IoT.js-API-Module.md [hunk 5]
```diff
@@ -2,48 +2,42 @@
...
+assert.equal(2, 2);
+```
 
-1. current working directory
-2. node_modules directory under current working directory
-3. `$HOME/node_modules`
-4. `$NODE_PATH/node_modules`
+**Loading a module**
 
+If a native module named `id` exists, load it and return.
```
Comment (bzsolt) [inline docs/api/IoT.js-API-Module.md]: @zherczeg Yes, it has a higher priority.

File: docs/api/IoT.js-API-Module.md (hunk 3 above)
Comment (bzsolt) [inline docs/api/IoT.js-API-Module.md]: Yes, it has a higher priority.

**Comment by LaszloLango** [comment]
Please rebase to the current master

**Comment by bzsolt** [comment]
@LaszloLango Done.

**Comment by daeyeon** [comment]
LGTM

**Comment by yichoi** [comment]
Docs Done!
